docs/reference/changelog.md

```diff
@@ -4,13 +4,85 @@ title: Changelog
+
+- Add new version of `Result::fromOption`, which allows non-string errors
+  ([#4272](https://github.com/darklang/dark/pull/4272)).
+
+- Add new version of `Bytes::base64Decode`, which fails properly on invalid
+  input ([#4287](https://github.com/darklang/dark/pull/4287)).
+
+- Bug fix: Dates were offset in Analysis by the number of hours away the user is
+  from GMT ([#4260](https://github.com/darklang/dark/pull/4260)).
+
+- Improved handling of 64-bit ints in editor
+  ([#4257](https://github.com/darklang/dark/pull/4257)).
+
+### Rewrite Dark in Dark
+
+Much of ApiSever (the backend that powers the editor) is being rewritten in
+Dark.
+
+- Expose more internal staticassets fns for uploads
+  ([#4278](https://github.com/darklang/dark/pull/4278)).
+
+- Ported Presence functionality to `dark-editor` canvas
+  ([#4294](https://github.com/darklang/dark/pull/4294)).
+
+### Work towards future improvements
+
+- Add StdLib fns for Tuples
+  ([#4311](https://github.com/darklang/dark/pull/4311)). While tuples are not
+  yet available in the Editor, we're making steady progress and hope to have
+  them available next release.
+
+- Provide more metadata to the client regarding standard library functions
+  ([#4277](https://github.com/darklang/dark/pull/4277)). In a follow up PR, the
+  editor will leverage this data to display additional contextual data around
+  functions being used.
+
+### Internal improvements
+
+- Given OCaml deprecation, start to define and use new client types to match F#
+  backend types ([#4254](https://github.com/darklang/dark/pull/4254)
+  [#4284](https://github.com/darklang/dark/pull/4284)
+  [#4288](https://github.com/darklang/dark/pull/4288)
+  [#4281](https://github.com/darklang/dark/pull/4281)
+  [#4282](https://github.com/darklang/dark/pull/4282)).
+
+- Continue migrating away from OCaml-compatible serializers
+  ([#4252](https://github.com/darklang/dark/pull/4252)
+  [#4256](https://github.com/darklang/dark/pull/4256)
+  [#4279](https://github.com/darklang/dark/pull/4279)).
+
+- Make build work on m1/m2 macs
+  ([#4237](https://github.com/darklang/dark/pull/4237)).
+
+- Make toplevel serialization output more obvious
+  ([#4331](https://github.com/darklang/dark/pull/4331)).
+
+- Update/improve 'running against prod' doc
+  ([#4300](https://github.com/darklang/dark/pull/4300)).
+
+- Allow admin fns for all admin canvases
+  ([#4271](https://github.com/darklang/dark/pull/4271)).
+
+- Smaller code-cleanup efforts
+  ([#4238](https://github.com/darklang/dark/pull/4238)
+  [#4285](https://github.com/darklang/dark/pull/4285)
+  [#4255](https://github.com/darklang/dark/pull/4255)
+  [#4291](https://github.com/darklang/dark/pull/4291)).
+
 ## Darklang Release 3 (July 1, 2022)
 
 Darklang Release 3 is focused primarily on taking advantage of the recent
 changes, including new stdlib functions, small languages changes, and sunsetting
 old features.
 
-## Language improvements
+### Language improvements
```

Comment:
I saw that we had inconsistent headers in the previous release, so I turned some h2s into h3s